### PR TITLE
[CN-632] Compatibility tests in Kind cluster

### PR DESCRIPTION
**Pull request by semihbkgr**
## Description

Run compatibility tests on `kind` cluster.

`GKE` cluster in #1053 is changed to `Kind`

### CODE REVIEW

**Comment by semihbkgr** [comment]
It is ready
https://github.com/hazelcast/hazelcast-platform-operator/actions/runs/8177743104

**Comment by hasancelik** [comment]
@semihbkgr with this PR. you are partially reverting your previous PR:
https://github.com/hazelcast/hazelcast-platform-operator/pull/1053/files
Could you mention it in the description?

File: .github/workflows/e2e-kind.yaml [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+on:
+  workflow_dispatch:
+  workflow_call:
+    inputs:
+      k8s-cluster-version:
+        description: "kubernetes cluster version"
+        required: false
+        type: string
+        default: "1.29"
```
Comment (cheels) [inline .github/workflows/e2e-kind.yaml:11]: What we will do after 3-4 months when 1.30 will be released? Will we update it all the time? Maybe it is worth to make it dynamically?

File: .github/workflows/k8s-compatibility-test.yaml [hunk 2]
```diff
@@ -2,18 +2,99 @@ name: (E2E) K8s Compatibility Tests
...
   run-test:
-    uses: ./.github/workflows/e2e-gke.yaml
+    uses: ./.github/workflows/e2e-kind.yaml
     secrets: inherit
     name: Run E2E tests
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions: ["1.29", "1.28", "1.27"]
+        kubernetes-versions: ["1.29", "1.28", "1.27", "1.26", "1.25"]
```
Comment (cheels) [inline .github/workflows/k8s-compatibility-test.yaml:98]: Same here, these values should be always updated manually.  What if can add the latest version here automatically?

**Review by hasancelik — APPROVED** [review]

File: .github/workflows/e2e-kind.yaml (hunk 1 above)
Comment (semihbkgr) [reply .github/workflows/e2e-kind.yaml:11]: I agree, it would be good to automate this process.
I created a separate task for that: CN-1165

**Review by cheels — APPROVED** [review]

**Pull request merged**